src/v/kafka/server/handlers/produce.cc

```diff
@@ -348,9 +355,18 @@ produce_topic(produce_ctx& octx, produce_request::topic& topic) {
             continue;
         }
 
+        if (materialized) {
+            partitions_dispatched.push_back(ss::now());
+            partitions_produced.push_back(
+              ss::make_ready_future<produce_response::partition>(
+                produce_response::partition{
+                  .partition_index = part.partition_index,
+                  .error_code = error_code::invalid_config}));
```

Comment:
`invalid_config` Kafka error code is never returned in ProduceResponse, we should find better suited error code

Reply:
I agree, hows this one?
``` 
// The request attempted to perform an operation on an invalid topic.
invalid_topic_exception = 17,
```